components/raftstore/src/coprocessor/region_info_accessor.rs

```diff
@@ -239,7 +290,7 @@ pub struct RegionCollector {
     regions: RegionsMap,
     // BTreeMap: data_end_key -> region_id
     region_ranges: RegionRangesMap,
-
+    region_activity: RegionActivityMap,
```

Comment:
I think we can use std::collections::BinaryHeap to keep top N regions, beside the RegionActivityMap. 
With help of BinaryHeap, we can easily keep a top N list of regions and then the sort will be much cheaper. Also it can reduce the memory usage as well. I think it's worth the effort. 

Or even better is we can wrap the hashmap and binaryheap into a struct and hide the details inside it. Here we only call add/delete.

Reply:
I think it can be done in another PR later though.

Reply:
Added a TODO.